### PR TITLE
feat(vector search): Add examples for mutate, PSC, and VPC

### PR DESCRIPTION
**Pull request by ericgribkoff**
## Checklist

**Readiness**

- [x] Yes, **merge** this PR after it is approved
- [ ] No, don't **merge** this PR after it is approved

**Style**

- [x] My sample follows the rules described for Terraform in the [Effective Samples style guide](https://googlecloudplatform.github.io/samples-style-guide/)
- [x] My sample follows the other requirements and best practices in the [Contributing
guide](https://github.com/terraform-google-modules/terraform-docs-samples/blob/main/CONTRIBUTING.md#other-requirements-and-best-practices)

**Testing**

- [x] I have performed tests described in the [Contributing guide](https://github.com/terraform-google-modules/terraform-docs-samples/blob/main/CONTRIBUTING.md):

   - [x] **[Tests](https://github.com/terraform-google-modules/terraform-docs-samples/blob/main/CONTRIBUTING.md#set-up-the-test-environment)** pass: `terraform apply`
   - [x] **[Lint](https://github.com/terraform-google-modules/terraform-docs-samples/blob/main/CONTRIBUTING.md#linting-and-formatting)** pass: `terraform fmt` check

**Intended location**

- [x] Yes, this sample will be (or already is) included on cloud.google.com
      Location(s):

- [ ] No, this sample won't be included on cloud.google.com
      Reason:

**API enablement**

- [ ] If the sample needs an API enabled to pass testing, I have added the service to the [Test setup file](https://github.com/terraform-google-modules/terraform-docs-samples/blob/main/test/setup/main.tf)

**Review**

- [ ] If this sample adds a new directory, I have added codeowners to the [CODEOWNERS file](https://github.com/terraform-google-modules/terraform-docs-samples/blob/main/.github/CODEOWNERS)

### CODE REVIEW

**Comment by glasnt** [comment]
Build failures: 

```
 Unable to remove Service Networking Connection, err: Error waiting for Delete Service Networking Connection: Error code 9, message: Failed to delete connection; Producer services (e.g. CloudSQL, Cloud Memstore, etc.) are still using this connection.
```

**Comment by ericgribkoff** [comment]
Thanks. Pushed two commits that should address the test failures (destroying `google_service_networking_connection` resources doesn't seem to play nicely with residual PSA resources, see commit message for more context).

**Comment by ericgribkoff** [comment]
Friendly ping on this

**Comment by glasnt** [comment]
New issue: 

```
Error: Error creating IndexEndpointDeployedIndex: googleapi: Error 400: There exists a DeployedIndex with same ID "deployed_index_id" either in failed state or being undeployed at the following IndexEndpoint: projects/189629635876/locations/us-central1/indexEndpoints/4010996427894292480. It will be deleted automatically, please retry again later or use a different ID.
```

From https://registry.terraform.io/providers/hashicorp/google/latest/docs/resources/vertex_ai_index_endpoint_deployed_index#deployed_index_id-1

> The ID must be unique within the project it is created in.

Due to CI (currently 4 threads, and this has five samples), all the deployed indexes in this sample must have unique names. I'll update each index to include this.

File: vertex_ai/index_endpoint_deploy_psc/main.tf [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+# [START aiplatform_deploy_index_endpoint_psc_sample]
+provider "google" {
+  region = "us-central1"
+}
+
+resource "google_vertex_ai_index_endpoint_deployed_index" "default" {
+  depends_on        = [google_vertex_ai_index_endpoint.default]
+  index_endpoint    = google_vertex_ai_index_endpoint.default.id
+  index             = google_vertex_ai_index.default.id
+  deployed_index_id = "deployed_index_id"
```
Comment (glasnt) [inline vertex_ai/index_endpoint_deploy_psc/main.tf:27]:
```suggestion
  deployed_index_id = "deployed_index_for_psc"
```

File: vertex_ai/index_endpoint_deploy_vpc/main.tf [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+# [START aiplatform_deploy_index_endpoint_vpc_sample]
+provider "google" {
+  region = "us-central1"
+}
+
+resource "google_vertex_ai_index_endpoint_deployed_index" "default" {
+  depends_on        = [google_vertex_ai_index_endpoint.default]
+  index_endpoint    = google_vertex_ai_index_endpoint.default.id
+  index             = google_vertex_ai_index.default.id
+  deployed_index_id = "deployed_index_id"
```
Comment (glasnt) [inline vertex_ai/index_endpoint_deploy_vpc/main.tf:27]:
```suggestion
  deployed_index_id = "deployed_index_for_vpc"
```

File: vertex_ai/index_endpoint_mutate/main.tf [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+# [START aiplatform_mutate_index_endpoint_sample]
+provider "google" {
+  region = "us-central1"
+}
+
+resource "google_vertex_ai_index_endpoint_deployed_index" "default" {
+  depends_on        = [google_vertex_ai_index_endpoint.default]
+  index_endpoint    = google_vertex_ai_index_endpoint.default.id
+  index             = google_vertex_ai_index.default.id
+  deployed_index_id = "deployed_index_id"
```
Comment (glasnt) [inline vertex_ai/index_endpoint_mutate/main.tf:27]:
```suggestion
  deployed_index_id = "deployed_index_for_mutate"
```

File: vertex_ai/index_endpoint_mutate_psc/main.tf [hunk 4]
```diff
@@ -0,0 +1,102 @@
...
+# [START aiplatform_mutate_index_endpoint_psc_sample]
+provider "google" {
+  region = "us-central1"
+}
+
+resource "google_vertex_ai_index_endpoint_deployed_index" "default" {
+  depends_on        = [google_vertex_ai_index_endpoint.default]
+  index_endpoint    = google_vertex_ai_index_endpoint.default.id
+  index             = google_vertex_ai_index.default.id
+  deployed_index_id = "deployed_index_id"
```
Comment (glasnt) [inline vertex_ai/index_endpoint_mutate_psc/main.tf:27]:
```suggestion
  deployed_index_id = "deployed_index_for_mutate_psc"
```

File: vertex_ai/index_endpoint_mutate_vpc/main.tf [hunk 5]
```diff
@@ -0,0 +1,120 @@
...
+# [START aiplatform_mutate_index_endpoint_vpc_sample]
+provider "google" {
+  region = "us-central1"
+}
+
+resource "google_vertex_ai_index_endpoint_deployed_index" "default" {
+  depends_on        = [google_vertex_ai_index_endpoint.default]
+  index_endpoint    = google_vertex_ai_index_endpoint.default.id
+  index             = google_vertex_ai_index.default.id
+  deployed_index_id = "deployed_index_id"
```
Comment (glasnt) [inline vertex_ai/index_endpoint_mutate_vpc/main.tf:27]:
```suggestion
  deployed_index_id = "deployed_index_for_mutate_vpc"
```

**Review by glasnt — APPROVED** [review]

**Comment by ericgribkoff** [comment]
Thanks @glasnt. The PSC tests in CI were failing with "Error waiting for Creating IndexEndpointDeployedIndex: Error code 5, message: A resource needed to create the ServiceAttachment could not be found: 'ci-tf-samples-2'. Please ensure that all input resources exist." I couldn't replicate this exact error, but the allowlist config was wrong (it should have been `project_id`). I updated the PR, and confirmed in a fresh GCP project that the PSC examples work. Can you please retrigger the CI? Thanks!

**Comment by ericgribkoff** [comment]
@msampathkumar Would you be able to trigger the CI build? I don't think I have proper permissions to trigger the cloud build.

**Comment by ericgribkoff** [comment]
Thanks @msampathkumar ! The CI passed. Can this be merged now?

**Comment by ericgribkoff** [comment]
Friendly ping on this

**Comment by glasnt** [comment]
The current error on this for index_endpoint_deploy_vpc:

```
Error: Error when reading or editing IndexEndpoint: googleapi: Error 400: The IndexEndpoint "projects/[PROJECT]/locations/us-central1/indexEndpoints/[INDEXENDPOINT]" has deployed or being-deployed DeployedIndex(s). Please undeploy all DeployedIndex(s) before deleting the IndexEndpoint.
```

I'm not sure if this is an eventual consistency issue, as the same resource association (index, endpoint, deployed_index) exists in the index_endpoint_deploy sample (already in the repo). 

index_endpoint_deploy_vpc is failing first before index_endpoint_deploy_psc has a chance to finish, and since CI fails early I can't confirm the state of index_endpoint_deploy_psc's tests. 

I'm going to re-run the tests again and confirm if it's the same issue. 

/gcbrun

**Comment by ericgribkoff** [comment]
The tests passed this time; not sure how to interpret that, as I'm having trouble locating the logs of the earlier failed run. My initial thought then was that the deployed endpoint hadn't been _fully_ deleted when the terraform teardown of the endpoint itself is attempted. This is probably what you were referring to with the eventual consistency possibility mentioned in the last comment, or is that a different possible cause?

**Comment by glasnt** [comment]
It's possibly an eventual consistency issue, since the last tests ran. If the current rebased-changes run clean, this sample will be automerged.

**Comment by glasnt** [comment]
@ericgribkoff this has been merged!

**Pull request merged**